interview-questions.md

```diff
@@ -8,50 +8,50 @@ Then, research the question to expand on your answer. Even if you feel you have
 
-Researched answer:
+Researched answer: The value, index, and array are the predefined parameters and the only required parameter is the value. The array is the array of the current element (value). 
 
 3. What is the difference between map and filter?
 
-Your answer:
+Your answer: Good question. The .map() method is iterating through an object, array, or string and returning a new array with the exact same length. The .filter() method iterates through an object, array, or string until the condition is satisified or the condition is met. 
 
-Researched answer:
+Researched answer: I was correct with my definitions. To add onto the .filter() method, the .filter() method needs something to evaluate such as a truthy or falsy, or a boolean (true or false) answer. 
 
 4. What is iteration?
 
-Your answer:
+Your answer: Good question. Iteration is the repitition over a string, number, or array that won't stop until the condition is satisfied and executed. 
 
-Researched answer:
+Researched answer: The process of repeating an asked upon action until that action is completed or satisifed by the condition set. 
 
 5. What is the difference between a class and an object?
 
-Your answer:
+Your answer: Great question. What I do know is that, a class is category with subcategories that has objects attached to it. A class also has a constructor which at this time I'm drawing a blank on. However, and object has key value pairs attached to them. While they cannot be called upon they are the placeholders or blueprint foundation for an object. 
 
-Researched answer:
+Researched answer: Objects are key value pairs that are stored in curly braces. They cannot be called upon without having dot notation to access the variable name and the key value pair. A class is the blueprint to the object. A class has three different keywords the classes name, the constructor, the this., and the new. The classes name which is PascalCased meaning the first letter is capital, for example, a class called PetNames, the p will be capitalized and the n. The second syntax is the constructor which is the access we need to create a new object and set values for those objects. Lastly, the new is the new object we are creating for the class. Also, classes and objects are reuseable.
 
 6. STRETCH: What is hoisting in JavaScript?
 
-Your answer:
+Your answer: That is a great question, and at this time I do not have an answer for you, can you please explain what that is and I will be sure to do research on this specific defintion. 
 
-Researched answer:
+Researched answer: A variable can be used before it was declared. This could be a letter is assigned a number. It is the default behavior for Javascript to move all declared variables to the top of the scope which means in the beginning inside a function. 
```

Comment:
Your researched answer is a perfect explanation.  Well done! This is a common interview question.